### ISSUE TITLE
[Request] Can we get an event for when the display frequency changes?

### ISSUE THREAD

**Issue by BOLL7708** [ValveSoftware/openvr#1335]
I want to keep track of the display frequency of the headset in a background application, and while I can poll `Prop_DisplayFrequency_Float` every second, it would be nice to have an `EVREventType` for this, like `VREvent_DisplayFrequencyChanged` or similar. This exists for other properties like `VREvent_IpdChanged` and `VREvent_TrackedDeviceRoleChanged` 🤔

I do see that `VREvent_SteamVRSectionSettingChanged` is triggered, but I cannot find any value in the event data that would tell me which specific setting was changed, unless I'm missing something. And other settings will trigger the same event, so only partially useful for me right now 🤷‍♂️ If I could tell it was the frequency that was changed I can load the property myself no problem.

For the moment I will just go with event above and load the frequency when that happens, so no panic 😅 But it would be nice to have!

**Comment by BOLL7708** [comment]
Was going to report this again, but did find my own issue, so I will add more information here.

This time around I monitored incoming ```EVREventType.VREvent_PropertyChanged``` events and matched ```VREvent_t.data.property.prop``` with ```ETrackedDeviceProperty.Prop_DisplayFrequency_Float```. But, even if the property is indeed changed when I load it, there is no event triggered for this either 🤷‍♂️

At this point it feels more like a missing feature than a feature request, to be honest.

Using a Valve Index, changing headset refresh rate at runtime through the SteamVR Settings. SteamVR version 1.11.13

**Comment by BOLL7708** [comment]
Ah, yet again I was going to post this but found my own issue from search. Hello past me, we meet again. As in the first post I notice again that `VREvent_SteamVRSectionSettingChanged` is triggered when changing this, but when debugging the data that call gives me in its `VREvent_t` doesn't seem to have any information as to what setting changed.

I actually forgot to implement what I mentioned in the first post, which I plan to do now, so I'll just load the display frequency on every settings change, as that's what I have to go on right now unless I want to run a continuous timer.

**Comment by BOLL7708** [comment]
> Hi, BOLL! Check VREvent_PropertyChanged with VREvent_t.data.property.prop==Prop_SecondsFromVsyncToPhotons_Float. Curently it doesn't work with Steam Link( #1800 ) but it works great in the case of Valve Index. I hope my answer is not too late :D

Thanks Sergio! Sorry for not replying sooner, I haven't been working on my OpenVR projects for a few years 😅 this as I've focused on a Twitch bot. I did star the email notification anticipating I would start this up sooner, but it wasn't until this week I decided to pause the bot development to update all my OpenVR applications, and as such this will be something I add support for now, so thanks for the ping! 😁
